astropy/modeling/fitting.py

```diff
@@ -1013,7 +1013,8 @@ def __call__(self, model, x, y, z=None, weights=None, *, inplace=False, **kwargs
         n = 0  # (allow recording no. of iterations when 0)
 
         # Perform the iterative fitting:
-        for n in range(1, self.niter + 1):
+        for i in range(1, self.niter + 1):
+            n=i
```

Comment:
shouldn't this be
```suggestion
        for _ in range(1, self.niter + 1):
```
?

Reply:
The numbers of loops executed is recorded and there is a possibility of breaking early, but you have to go all the way down to line 1109 (or 1108 in `main`) to see that. I would rename `n` to `niter` and replace
```suggestion
        for _ in range(self.niter):
            niter += 1
```
Then it would be clearer that `niter` is a counter that will be used after the loop, not just a loop variable.